src/rust/engine/src/lib.rs

```diff
@@ -92,8 +92,8 @@ impl RawNode {
           (RawStateTag::Empty as u8, externs::create_exception("No value")),
         Some(Ok(v)) =>
           (RawStateTag::Return as u8, v),
-        Some(Err(Failure::Throw(msg))) =>
-          (RawStateTag::Throw as u8, msg),
+        Some(Err(Failure::Throw(exc, _))) =>
```

Comment:
Carrying the rendered backtrace around like this feels redundant, although it's probably not a huge deal. Might be cleaner to have a python function to render an exception?